apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx

```diff
@@ -981,13 +984,16 @@ export class DetailViewContents extends React.Component {
   }
 
   showPrincipalApprovalTable = () => {
-    return (
-      this.props.applicationData.principal_approval_state || ''
-    ).startsWith('Complete');
+    return this.props.applicationData.principal_approval_state?.startsWith(
+      'Complete'
+    );
```

Comment:
for future consideration: it would be better if application behavior did not depend on UI string contents. this approach is fragile in case we (1) change the strings or (2) start to translate them. my recommendation would be to use two separate variables, one to control application behavior ('incomplete', 'complete', etc) and another containing the UI string.

Reply:
oh, interesting. yes, comparing for equality against that constant seems like a nice improvement!